git_helper.go

```diff
@@ -0,0 +1,278 @@
+	}
+
+	clonePath := filepath.Join(g.tempDir, repoName)
+
+	// Perform sparse clone
+	args := []string{"clone", "--filter=blob:none", "--no-checkout"}
+	args = append(args, repoURL, clonePath)
+	cmd := exec.Command("git", args...)
+
+	output, err := cmd.CombinedOutput()
+	if err != nil {
+		return "", fmt.Errorf("git clone failed: %w\nOutput: %s", err, output)
+	}
+
+	// Initialize sparse checkout
+	cmd = exec.Command("git", "sparse-checkout", "init", "--cone")
+	cmd.Dir = clonePath
+
+	output, err = cmd.CombinedOutput()
+	if err != nil {
+		os.RemoveAll(clonePath)
+
+		return "", fmt.Errorf("git sparse-checkout init failed: %w\nOutput: %s", err, output)
+	}
+
+	g.clonedRepos[repoURL] = clonePath
+
+	return clonePath, nil
+}
+
+// Checks out a specific file from the cloned repository.
+func (g *gitHelper) checkoutFile(repoPath string, filePath string) error {
+	// First, add the file to sparse-checkout
+	dir := filepath.Dir(filePath)
+	if dir != "." && dir != "" {
+		cmd := exec.Command("git", "sparse-checkout", "add", dir)
+		cmd.Dir = repoPath
+
+		_, err := cmd.CombinedOutput()
+		if err != nil {
+			// Try to add the specific file if directory fails
+			cmd = exec.Command("git", "sparse-checkout", "add", filePath)
+			cmd.Dir = repoPath
+
+			output, err := cmd.CombinedOutput()
+			if err != nil {
+				return fmt.Errorf("git sparse-checkout add failed: %w\nOutput: %s", err, output)
+			}
+		}
+	}
+
+	// Checkout the file
+	cmd := exec.Command("git", "checkout", "HEAD", "--", filePath)
+	cmd.Dir = repoPath
+
+	output, err := cmd.CombinedOutput()
+	if err != nil {
+		// File might not exist in the repository
+		return fmt.Errorf("git checkout failed: %w\nOutput: %s", err, output)
+	}
+
+	return nil
+}
+
+// Checks if a file exists in a Git repository by attempting to check it out.
+func (g *gitHelper) fileExistsInRepo(repoURL string, filePath string) (bool, string, error) {
+	// Clone the repository if not already cloned
+	clonePath, err := g.cloneRepo(repoURL)
+	if err != nil {
+		return false, "", err
+	}
+
+	// Try to checkout the file
+	err = g.checkoutFile(clonePath, filePath)
+	if err != nil {
+		// File doesn't exist in the repository
+		return false, "", err
+	}
+
+	// File exists and has been checked out
+	localPath := filepath.Join(clonePath, filePath)
+	if _, err := os.Stat(localPath); err != nil {
+		// File was supposedly checked out but doesn't exist - this is an error
+		return false, "", fmt.Errorf("file was checked out but not found at %s: %w", localPath, err)
+	}
+
+	return true, localPath, nil
+}
+
+// Removes all temporary directories and cloned repositories.
+func (g *gitHelper) cleanup() error {
+	if g.tempDir != "" {
+		return os.RemoveAll(g.tempDir)
+	}
+
+	return nil
+}
+
+// Checks if an URL is a generic Git repository URL.
+// Returns false for supported hosting platforms (GitHub, GitLab, Bitbucket)
+// which have web interfaces and should not use local Git cloning.
+func isGitURL(u *url.URL) bool {
```

Comment:
Commenting on this function, but it's a general  remark.

`isGitURL`, `isGitRepo`, are misleading. We should pick an alternative way of expressing the concept: They're always git repos in all cases, and it might be confusing to follow the code with this naming.